aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -144,6 +277,169 @@ impl AptosVM {
         self
     }
 
+    pub(crate) fn new_session<'r, S: AptosMoveResolver>(
+        &self,
+        resolver: &'r S,
+        session_id: SessionId,
+    ) -> SessionExt<'r, '_> {
+        self.move_vm.new_session(resolver, session_id)
+    }
+
+    fn get_gas_parameters(
+        &self,
+        log_context: &AdapterLogSchema,
+    ) -> Result<&AptosGasParameters, VMStatus> {
+        self.gas_params.as_ref().map_err(|err| {
+            let msg = format!("VM Startup Failed. {}", err);
+            speculative_error!(log_context, msg.clone());
+            VMStatus::error(StatusCode::VM_STARTUP_FAILURE, Some(msg))
+        })
+    }
+
+    fn get_storage_gas_parameters(
+        &self,
+        log_context: &AdapterLogSchema,
+    ) -> Result<&StorageGasParameters, VMStatus> {
+        self.storage_gas_params.as_ref().map_err(|err| {
+            let msg = format!("VM Startup Failed. {}", err);
+            speculative_error!(log_context, msg.clone());
+            VMStatus::error(StatusCode::VM_STARTUP_FAILURE, Some(msg))
+        })
+    }
+
+    fn check_gas(
+        &self,
```

Comment:
similarly, self only seems used for get_gas_parameters and gas_feature version, which can be provided as parameters.

Reply:
Good point, I created a `gas.rs` helper file, will move gas related things there